File: week-08/day-2/mysqltodo/src/main/java/com/greenfoxacademy/mysqltodo/services/TodoService.java
package com.greenfoxacademy.mysqltodo.services;

import com.greenfoxacademy.mysqltodo.models.Todo;

import java.util.List;

public interface TodoService {
  
  List<Todo> findTodos();
  List<Todo> findActiveTodos();
  Todo findTodoById(String id);
  void saveTodo(Todo todo);
  void deleteTodo(String id);
  List<Todo> findListFromUserInput(String userInput);
}

File: week-10/day-2/TodoApp/TodoApp/Services/TodoService.cs
using System.Collections.Generic;
using TodoApp.Models;
using TodoApp.Repositories;

namespace TodoApp.Services
{
    public class TodoService : ITodoService
    {

        private readonly ITodoRepository todoRepository;

        public TodoService(ITodoRepository todoRepository)
        {
            this.todoRepository = todoRepository;
        }

        private List<Todo> GetAllTodos()
        {
            return todoRepository.FindAllTodos();
        }

        public void SaveTodo(Todo todo)
        {
            todoRepository.SaveTodo(todo);
        }

        public void DeleteTodoById(int id)
        {
            var todo = GetTodoById(id);
            if (todo != null)
            {
                todoRepository.DeleteTodo(todo);
            }
        }

        public Todo GetTodoById(int id)
        {
            return todoRepository.FindTodoById(id);
        }

        public List<Todo> GetTodoList(string search)
        {
            return search == null ? GetAllTodos() : todoRepository.FindTodosByTitle(search);
        }
    }
}
